components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidator.java

```diff
@@ -0,0 +1,211 @@
+    private static final String SP_NAME_ID = SP_CATEGORY + "/sp-name";
+    private static final String USERNAME_ID = USER_CATEGORY + "/username";
+    private static final String USER_STORE_ID = USER_CATEGORY + "/user-store-domain";
+    private static final String USER_TENANT_DOMAIN_ID = USER_CATEGORY + "/user-tenant-domain";
+    private static final String SCOPE_ID = SCOPE_CATEGORY + "/scope-name";
+
+    private Log log = LogFactory.getLog(XACMLScopeValidator.class);
+
+    @Override
+    public boolean validateScope(AccessTokenDO accessTokenDO, String resource) throws IdentityOAuth2Exception {
+
+        if (accessTokenDO.getAuthzUser() == null) {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("There is no authorized user for client id %s ",
+                        accessTokenDO.getConsumerKey()));
+            }
+            return false;
+        }
+
+        boolean isValidated = false;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("In XACML based scope validation flow for access token of consumer key : %s ",
+                        accessTokenDO.getConsumerKey()));
+            }
+            String consumerKey = accessTokenDO.getConsumerKey();
+            OAuthAppDO authApp = OAuth2Util.getAppInformationByClientId(consumerKey);
+
+            RequestDTO requestDTO = createRequestDTO(accessTokenDO, authApp, resource);
+            RequestElementDTO requestElementDTO = PolicyCreatorUtil.createRequestElementDTO(requestDTO);
+            String requestString = PolicyBuilder.getInstance().buildRequest(requestElementDTO);
+
+            if (log.isDebugEnabled()) {
+                log.debug("XACML scope validation request :\n" + requestString);
+            }
+            FrameworkUtils.startTenantFlow(accessTokenDO.getAuthzUser().getTenantDomain());
+            String responseString = OAuth2ServiceComponentHolder.getEntitlementService().getDecision(requestString);
+            if (log.isDebugEnabled()) {
+                log.debug("XACML scope validation response :\n" + responseString);
+            }
+            String validationResponse = evaluateXACMLResponse(responseString);
+            if (RULE_EFFECT_NOT_APPLICABLE.equalsIgnoreCase(validationResponse)) {
+                log.warn(String.format(
+                        "No applicable rule for service provider '%s@%s'. Add an validating policy (or unset Scope " +
+                                "Validation using XACMLScopeValidator) to fix this warning.",
+                        authApp.getApplicationName(), OAuth2Util.getTenantDomainOfOauthApp(authApp)));
+                isValidated = true;
+            } else if (RULE_EFFECT_PERMIT.equalsIgnoreCase(validationResponse)) {
+                isValidated = true;
+            }
+        } catch (InvalidOAuthClientException e) {
+            log.error(String.format("Exception occurred when getting app information for client id %s ",
```

Comment:
As I know the idea plugin indicates warnings for the String.format();
Can you check that issue as well?